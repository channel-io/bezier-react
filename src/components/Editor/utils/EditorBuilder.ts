/* External dependencies */
import { EditorView, DirectEditorProps } from 'prosemirror-view'
import {
  EditorState,
  Plugin,
  Transaction,
} from 'prosemirror-state'
import {
  Schema,
  NodeSpec,
  MarkSpec,
  Node as ProsemirrorNode,
} from 'prosemirror-model'
import { InputRule } from 'prosemirror-inputrules'
import { keymap } from 'prosemirror-keymap'
import {
  chainCommands,
  createParagraphNear,
  liftEmptyBlock,
  splitBlock,
} from 'prosemirror-commands'
import { mapValues, last } from 'lodash-es'

/* Internal dependencies */
import { NodeType } from '../Editor.types'
import { ReactNodeView, ReactNodeViewProvider } from './ReactNodeView'
import { getNodeFromSelection } from './SelectionUtils'

type Dispatch = (tr: Transaction<any>) => void

type TriggerCallback = (active: boolean, searchWord: string, state: EditorState) => void

const newLineCommand = chainCommands(createParagraphNear, liftEmptyBlock, splitBlock)

export default class EditorBuilder {
  schema: Schema | undefined

  plugins: Plugin[] = []

  editor: EditorView | null = null

  nodes: { [key: string]: NodeSpec } = {
    doc: {
      content: 'block+',
    },
    paragraph: {
      content: 'inline*',
      group: 'block',
      parseDOM: [
        { tag: 'p' },
        { tag: 'span:empty' },
        /**
         * TODO: 주석 번역
         * NOTE: 두 개 이상의 연속되는 br 은 온전히 하나의 줄을 차지할 수 있으므로,
         * p 태그로 변환해도 됩니다. 하지만 단일 br 은 단순 문단 분리의 동작을 수행해야 하는데,
         * 현재의 Prosemirror parseRule 에서는 해당 동작을 구현할 수 있는 방법이 없습니다.
         * 그래서 임시로 br 태그를 직접 렌더링하고, PMNodesToBlocksVisitor 에서 처리하도록 수정합니다.
         *
         * 이 방법의 문제는, br 로 연결된 모든 텍스트가 단일 paragraph로 인식되는 것으로,
         * 예기치 않은 동작을 할 수 있습니다.
         *
         * 아래 옵션은 연속하는 br 을 p 로 변환하는 parseRule 입니다.
         * { tag: 'br + br' },
         */
      ],
      toDOM: () => (['p', 0]),
    },
    text: {
      group: 'inline',
      marks: '_',
    },
  }

  marks: { [key: string]: MarkSpec } = {}

  keyMaps: { [key: string]: (state: EditorState, dispatch: Dispatch) => void } = {}

  nodeViews: { [key: string]: new(node: ProsemirrorNode, provider: ReactNodeViewProvider) => ReactNodeView } = {}

  inputRules: InputRule[] = []

  widgetItems: any[] = []

  triggers: { [key: string]: TriggerCallback } = {}

  onUpdates: Set<(state: EditorState) => void> = new Set()

  createSchema() {
    const { nodes, marks } = this

    return new Schema({
      nodes,
      marks,
    })
  }

  createState(schema?: Schema, doc?: ProsemirrorNode) {
    const { plugins } = this

    return EditorState.create({
      doc,
      schema,
      plugins,
    })
  }

  build(
    target: HTMLElement,
    provider: ReactNodeViewProvider,
    schema: Schema,
    props: Partial<DirectEditorProps>,
    doc?: ProsemirrorNode,
  ) {
    const { nodeViews } = this

    this.plugins.push(keymap({
      ...this.keyMaps,
      Enter: newLineCommand,
    }))

    this.editor = new EditorView(target, {
      ...props,
      state: this.createState(schema, doc),
      nodeViews: mapValues(nodeViews, (NodeView) => (node: ProsemirrorNode) => new NodeView(node, provider)),
    })

    this.schema = schema
  }

  updateTrigger(state: EditorState) {
    const triggerWords = Object.keys(this.triggers)
    const { from, to, $from: { nodeBefore } } = state.selection
    const { parentNode } = getNodeFromSelection(state)

    if (
      (from === to) &&
      nodeBefore?.isText &&
      parentNode &&
      (parentNode.type.name !== NodeType.ListItem) &&
      to
    ) {
      const { text } = nodeBefore
      const options = text?.split(/(\s+)/g) || []
      const lastString = last(options) || ''

      triggerWords.forEach(trigger => {
        if (lastString.startsWith(trigger)) {
          this.triggers[trigger](true, lastString, state)
          return
        }

        this.triggers[trigger](false, '', state)
      })
      return
    }

    triggerWords.forEach(trigger => {
      this.triggers[trigger](false, '', state)
    })
  }

  updateState(state: EditorState) {
    this.editor!.updateState(state)
    this.onUpdates.forEach(update => {
      update(state)
    })
    this.updateTrigger(state)
  }

  /**
   * TODO: 주석 번역
   * NOTE: 새로운 state 를 만들어 update 하면 각 Plugin 에서 update 를 추적할 수 없기 때문에,
   * 명시적으로 의미 없는 update 를 한 번 더 발생시켜 update 를 알림
   */
  reset() {
    this.updateState(this.createState(this.schema))

    if (this.editor) {
      this.updateState(this.editor.state)
    }
  }

  getSchema() {
    return this.editor?.state.schema
  }

  addTriggerCallback(trigger: string, callback: TriggerCallback) {
    this.triggers[trigger] = callback
  }

  removeTriggerCallback(trigger: string) {
    delete this.triggers[trigger]
  }

  addUpdateCallback(callback: (state: EditorState) => void) {
    this.onUpdates.add(callback)
  }

  removeUpdateCallback(callback: (state: EditorState) => void) {
    this.onUpdates.delete(callback)
  }
}
