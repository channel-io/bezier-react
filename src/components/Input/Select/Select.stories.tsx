/* External dependencies */
import React from 'react'
import base from 'paths.macro'

/* Internal dependencies */
import {
  getObjectFromEnum,
  getTitle,
} from '../../../utils/storyUtils'
import Select from './Select'
import { SelectSize } from './Select.types'

export default {
  title: getTitle(base),
  component: Select,
  argTypes: {
    size: {
      control: {
        type: 'radio',
        options: getObjectFromEnum(SelectSize),
      },
    },
  },
}

const Template = (args) => (
  <Select {...args} />
)

export const Primary = Template.bind({})
Primary.args = {
  placeholder: 'hello',
  text: 'hello',
  iconComponent: 'calendar',
  disabled: false,
  withoutChevron: false,
  hasError: false,
  size: SelectSize.M,
}
