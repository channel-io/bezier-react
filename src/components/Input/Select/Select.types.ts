/* Internal dependencies */
import { ChildrenComponentProps } from '../../../types/ComponentProps'
import { OverlayProps } from '../../Overlay'
import { IconName } from '../../Icon'

export enum SelectSize {
  XL = 'XL',
  L = 'L',
  M = 'M',
  S = 'S',
}

export interface SelectRef {
  handleClickTrigger(): void
  handleHideDropdown(): void
  getDOMNode(): Element | Text | null
}

interface SelectProps extends ChildrenComponentProps {
  triggerTestId?: string
  dropdownTestId?: string
  size?: SelectSize
  disabled?: boolean
  defaultFocus?: boolean
  placeholder?: string
  iconComponent?: IconName | React.ReactNode
  text?: string
  withoutChevron?: boolean
  dropdownContainer?: HTMLElement | null
  dropdownPosition?: OverlayProps['position']
  hasError?: boolean
}

export default SelectProps
