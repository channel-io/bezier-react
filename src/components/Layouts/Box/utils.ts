/* External dependencies */
import { CSSObject } from 'styled-components'
import { isNil, isNumber, isString, omit } from 'lodash-es'

/* Internal dependencies */
import { Foundation } from 'Foundation'
import { omittedPropKeys } from './const'

/**
 * @link https://styled-components.com/docs/advanced#style-objects
 * @description Style Objects로 prop을 전달받으면서 불필요한 prop도 함께 섞여 들어옵니다. 이를 삭제합니다.
 */
export function getSanitizedCSSObject(props: CSSObject) {
  return omit(props, omittedPropKeys)
}

export function convertThemeProps(foundation?: Foundation, ...values: unknown[]) {
  for (const index in values) {
    if (Object.prototype.hasOwnProperty.call(values, index)) {
      const eachValue = values[index]
      if (!isNil(eachValue)) {
        if (isString(eachValue)) {
          return foundation?.theme?.[eachValue] ?? eachValue
        }
      }
    }
  }
  return undefined
}

export function convertSpacingProps(foundation?: Foundation, ...values: unknown[]) {
  for (const index in values) {
    if (Object.prototype.hasOwnProperty.call(values, index)) {
      const eachValue = values[index]
      if (!isNil(eachValue)) {
        if (isNumber(eachValue)) {
          return eachValue === 0 ? eachValue : `${eachValue}px`
        }
        if (isString(eachValue)) {
          return foundation?.spacing?.[eachValue] ?? eachValue
        }
      }
    }
  }
  return undefined
}
