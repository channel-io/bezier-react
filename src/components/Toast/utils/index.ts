import { css } from '../../../foundation'
import { Appearance, IconColor, Placement } from '../Toast.types'

function getIconColor(appearance: Appearance): string {
  switch (appearance) {
    case Appearance.Success:
      return IconColor.Success
    case Appearance.Warning:
      return IconColor.Warning
    case Appearance.Error:
      return IconColor.Error
    case Appearance.Info:
    default:
      return IconColor.Info
  }
}

function placements(placement: Placement) {
  switch (placement) {
    case Placement.TopCenter:
      return css`
          top: 0;
          left: 50%;
          transform: translateX(-50%);
        `
    case Placement.TopRight:
      return css`
          top: 0;
          right: 0;
        `
    case Placement.TopLeft:
      return css`
          top: 0;
          left: 0;
        `
    case Placement.BottomCenter:
      return css`
          bottom: 0;
          left: 50%;
          transform: translateX(-50%);
        `
    case Placement.BottomRight:
      return css`
          right: 0;
          bottom: 0;
        `
    case Placement.BottomLeft:
    default:
      return css`
        bottom: 0;
        left: 0;
      `
  }
}

const initPosition = (placement: Placement, isXPosition: boolean) => {
  switch (placement) {
    case Placement.TopLeft:
      return isXPosition ? '-120%' : '0'
    case Placement.TopCenter:
      return isXPosition ? '0' : '-120%'
    case Placement.TopRight:
      return isXPosition ? '120%' : '0'
    case Placement.BottomLeft:
      return isXPosition ? '-120%' : '0'
    case Placement.BottomCenter:
      return isXPosition ? '0' : '120%'
    case Placement.BottomRight:
      return isXPosition ? '120%' : '0'
    default:
      return '0'
  }
}

export {
  getIconColor,
  placements,
  initPosition,
}
