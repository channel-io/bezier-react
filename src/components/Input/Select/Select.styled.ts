/* Internal dependencies */
import {
  styled,
  css,
} from '../../../foundation'
import DisabledOpacity from '../../../constants/DisabledOpacity'
import InjectedInterpolation from '../../../types/InjectedInterpolation'
import { Overlay } from '../../Overlay'
import {
  inputWrapperStyle,
  focusedInputWrapperStyle,
  erroredInputWrapperStyle,
} from '../constants/InputWrapperStyle'
import { SelectSize } from './Select.types'

interface ContainerProps {
  interpolation?: InjectedInterpolation
}

export const Container = styled.div<ContainerProps>`
  position: relative;

  ${({ interpolation }) => interpolation};
`

interface TriggerProps {
  focus: boolean
  error: boolean
  disabled: boolean
  size: SelectSize
}

function selectSizeConverter(size: SelectSize) {
  switch (size) {
    case SelectSize.XL:
      return css`
        height: 54px;
      `
    case SelectSize.L:
      return css`
        height: 44px;
      `
    case SelectSize.S:
      return css`
        height: 28px;
      `
    case SelectSize.M:
    default:
      return css`
        height: 36px;
      `
  }
}

export const Trigger = styled.div<TriggerProps>`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-between;
  ${({ size }) => selectSizeConverter(size)};
  padding: 8px 12px;
  color: ${({ foundation }) => foundation?.theme?.['txt-black-darker']};
  ${({ foundation }) => foundation?.rounding?.round8};
  ${inputWrapperStyle};
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  user-select: none;
  background-color: ${({ foundation }) => foundation?.theme?.['bg-grey-lightest']};

  ${({ foundation }) => foundation?.transition?.getTransitionsCSS(['background-color', 'box-shadow'])};

  &:hover {
    background-color: ${({ foundation }) => foundation?.theme?.['bg-grey-lighter']};
  }

  ${({ focus }) => focus && css`
    ${focusedInputWrapperStyle};
    background-color: ${({ foundation }) => foundation?.theme?.['bg-grey-lighter']};
  `};

  ${({ error }) => error && erroredInputWrapperStyle};

  ${({ disabled }) => disabled && css`
    opacity: ${DisabledOpacity};
  `};
`

export const MainContentWrapper = styled.div`
  display: flex;
  align-items: center;
`

export const Dropdown = styled(Overlay)`
  width: 100%;
  min-width: 200px;
  min-height: 42px;
  max-height: 640px;

  ${({ foundation }) => foundation?.rounding?.round8};
  ${({ foundation }) => foundation?.elevation?.ev3()};
`
