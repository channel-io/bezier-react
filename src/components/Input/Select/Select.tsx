/* External dependencies */
import React, {
  useState,
  useCallback,
  useMemo,
  useRef,
  forwardRef,
  Ref,
  useImperativeHandle,
} from 'react'

/* Internal dependencies */
import {
  Icon,
  IconSize,
  isIconName,
} from '../../Icon'
import { Text } from '../../Text'
import { OverlayPosition } from '../../Overlay'
import { Typography } from '../../../foundation'
import SelectProps, {
  SelectRef,
  SelectSize,
} from './Select.types'
import * as Styled from './Select.styled'

export const SELECT_CONTAINER_TEST_ID = 'bezier-react-select-container'
export const SELECT_TRIGGER_TEST_ID = 'bezier-react-select-trigger'
export const SELECT_DROPDOWN_TEST_ID = 'bezier-react-select-dropdown'

function Select(
  {
    testId = SELECT_CONTAINER_TEST_ID,
    triggerTestId = SELECT_TRIGGER_TEST_ID,
    dropdownTestId = SELECT_DROPDOWN_TEST_ID,
    as,
    style,
    className,
    interpolation,
    size = SelectSize.M,
    disabled = false,
    defaultFocus = false,
    placeholder = '',
    iconComponent,
    text,
    withoutChevron = false,
    hasError = false,
    dropdownContainer,
    dropdownPosition = OverlayPosition.BottomLeft,
    children,
  }: SelectProps,
  forwardedRef: Ref<SelectRef>,
) {
  const containerRef = useRef<HTMLDivElement>(null)
  const triggerRef = useRef<HTMLDivElement>(null)

  const [isDropdownOpened, setIsDropdownOpened] = useState(defaultFocus)

  const LeftComponent = useMemo(() => {
    if (isIconName(iconComponent)) {
      return (
        <Icon
          name={iconComponent}
          size={IconSize.XS}
          marginRight={6}
        />
      )
    }

    return iconComponent
  }, [iconComponent])

  const handleClickTrigger = useCallback(() => {
    if (!disabled) {
      setIsDropdownOpened(prevState => !prevState)
    }
  }, [disabled])

  const handleHideDropdown = useCallback(() => {
    setIsDropdownOpened(false)
  }, [])

  const getDOMNode = useCallback(() => triggerRef.current, [])

  useImperativeHandle(forwardedRef, () => ({
    handleClickTrigger,
    handleHideDropdown,
    getDOMNode,
  }))

  return (
    <Styled.Container
      data-testid={testId}
      ref={containerRef}
      style={style}
      className={className}
      interpolation={interpolation}
    >
      <Styled.Trigger
        data-testid={triggerTestId}
        as={as}
        ref={triggerRef}
        size={size}
        focus={isDropdownOpened && !disabled}
        error={hasError}
        disabled={disabled}
        onClick={handleClickTrigger}
      >
        <Styled.MainContentWrapper>
          { LeftComponent }
          <Text typo={Typography.Size14}>
            { text || placeholder }
          </Text>
        </Styled.MainContentWrapper>
        { !withoutChevron && (
          <Icon
            name={isDropdownOpened ? 'chevron-up' : 'chevron-down'}
            size={IconSize.XS}
            marginLeft={6}
          />
        ) }
      </Styled.Trigger>

      <Styled.Dropdown
        testId={dropdownTestId}
        withTransition
        show={isDropdownOpened && !disabled}
        marginY={6}
        target={triggerRef.current}
        container={dropdownContainer || containerRef.current}
        position={dropdownPosition}
        onHide={handleHideDropdown}
      >
        { children }
      </Styled.Dropdown>
    </Styled.Container>
  )
}

export default forwardRef(Select)
