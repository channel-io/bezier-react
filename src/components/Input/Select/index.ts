import Select from './Select'
import { SelectSize } from './Select.types'
import type SelectProps from './Select.types'

export type {
  SelectProps,
}

export {
  Select,
  SelectSize,
}
