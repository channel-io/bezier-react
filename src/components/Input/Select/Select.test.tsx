/* External dependencies */
import React from 'react'
import userEvent from '@testing-library/user-event'

/* Internal dependencies */
import { render } from '../../../utils/testUtils'
import Select, {
  SELECT_CONTAINER_TEST_ID,
  SELECT_TRIGGER_TEST_ID,
  SELECT_DROPDOWN_TEST_ID,
} from './Select'
import SelectProps, { SelectSize } from './Select.types'

describe('Select Test >', () => {
  let props: SelectProps

  beforeEach(() => {
    const { body } = document
    body.innerHTML = ''

    const rootElement = document.createElement('div')
    rootElement.setAttribute('id', 'root')

    body.appendChild(rootElement)
  })

  const renderSelect = (optionProps?: Partial<SelectProps>) => render(<Select {...props} {...optionProps} />)

  describe('Default Style >', () => {
    it('Snapshot >', () => {
      const { container } = renderSelect()
      expect(container.firstChild).toMatchSnapshot()
    })

    it('Container >', () => {
      const { getByTestId } = renderSelect()
      const defaultSelect = getByTestId(SELECT_CONTAINER_TEST_ID)

      expect(defaultSelect).toHaveStyle('position: relative;')
    })

    it('Trigger >', () => {
      const { getByTestId } = renderSelect()
      const defaultSelectTrigger = getByTestId(SELECT_TRIGGER_TEST_ID)

      expect(defaultSelectTrigger).toHaveStyle('display: flex;')
      expect(defaultSelectTrigger).toHaveStyle('align-items: center;')
      expect(defaultSelectTrigger).toHaveStyle('justify-content: space-between;')
      expect(defaultSelectTrigger).toHaveStyle('padding: 8px 12px;')
      expect(defaultSelectTrigger).toHaveStyle('cursor: pointer;')
    })

    it('Dropdown >', () => {
      const { getByTestId } = renderSelect()

      const defaultSelectTrigger = getByTestId(SELECT_TRIGGER_TEST_ID)
      userEvent.click(defaultSelectTrigger)

      const defaultSelectDropdown = getByTestId(SELECT_DROPDOWN_TEST_ID)

      expect(defaultSelectDropdown).toHaveStyle('width: 100%;')
      expect(defaultSelectDropdown).toHaveStyle('min-width: 200px;')
      expect(defaultSelectDropdown).toHaveStyle('min-height: 42px;')
      expect(defaultSelectDropdown).toHaveStyle('max-height: 640px;')
    })
  })

  describe('SelectSize >', () => {
    it('Size only effects to trigger style >', () => {
      const XL_CONTAINER_TEST_ID = `${SELECT_CONTAINER_TEST_ID}_XL`
      const XL_TRIGGER_TEST_ID = `${SELECT_TRIGGER_TEST_ID}_XL`
      const XL_DROPDOWN_TEST_ID = `${SELECT_DROPDOWN_TEST_ID}_XL`
      const L_CONTAINER_TEST_ID = `${SELECT_CONTAINER_TEST_ID}_L`
      const L_TRIGGER_TEST_ID = `${SELECT_TRIGGER_TEST_ID}_L`
      const L_DROPDOWN_TEST_ID = `${SELECT_DROPDOWN_TEST_ID}_L`

      const { getByTestId: getByTestIdForXL } = renderSelect({
        size: SelectSize.XL,
        testId: XL_CONTAINER_TEST_ID,
        triggerTestId: XL_TRIGGER_TEST_ID,
        dropdownTestId: XL_DROPDOWN_TEST_ID,
      })
      const { getByTestId: getByTestIdForL } = renderSelect({
        size: SelectSize.L,
        testId: L_CONTAINER_TEST_ID,
        triggerTestId: L_TRIGGER_TEST_ID,
        dropdownTestId: L_DROPDOWN_TEST_ID,
      })

      // Select Size.XL
      const xlSelectContainer = getByTestIdForXL(XL_CONTAINER_TEST_ID)
      const xlSelectTrigger = getByTestIdForXL(XL_TRIGGER_TEST_ID)
      // Open Dropdown
      userEvent.click(xlSelectTrigger)
      const xlSelectDropdown = getByTestIdForXL(XL_DROPDOWN_TEST_ID)
      // Close Dropdown
      userEvent.click(xlSelectTrigger)
      const xlContainerStyle = window.getComputedStyle(xlSelectContainer)
      const xlDropdownStyle = window.getComputedStyle(xlSelectDropdown)

      // Select Size.L
      const lSelectContainer = getByTestIdForL(L_CONTAINER_TEST_ID)
      const lSelectTrigger = getByTestIdForL(L_TRIGGER_TEST_ID)
      // Open Dropdown
      userEvent.click(lSelectTrigger)
      const lSelectDropdown = getByTestIdForL(L_DROPDOWN_TEST_ID)
      // Close Dropdown
      userEvent.click(lSelectTrigger)
      const lContainerStyle = window.getComputedStyle(lSelectContainer)
      const lDropdownStyle = window.getComputedStyle(lSelectDropdown)

      // Container
      expect(xlContainerStyle.getPropertyValue('display')).toEqual(lContainerStyle.getPropertyValue('display'))

      // Dropdown
      expect(xlDropdownStyle.getPropertyValue('width')).toEqual(lDropdownStyle.getPropertyValue('width'))
      expect(xlDropdownStyle.getPropertyValue('min-width')).toEqual(lDropdownStyle.getPropertyValue('min-width'))
      expect(xlDropdownStyle.getPropertyValue('min-height')).toEqual(lDropdownStyle.getPropertyValue('min-height'))
      expect(xlDropdownStyle.getPropertyValue('max-height')).toEqual(lDropdownStyle.getPropertyValue('max-height'))
    })

    it('Size S >', () => {
      const { getByTestId } = renderSelect({ size: SelectSize.S })
      const defaultSelectDropdown = getByTestId(SELECT_TRIGGER_TEST_ID)

      expect(defaultSelectDropdown).toHaveStyle('height: 28px;')
    })

    it('Size M >', () => {
      const { getByTestId } = renderSelect({ size: SelectSize.M })
      const defaultSelectDropdown = getByTestId(SELECT_TRIGGER_TEST_ID)

      expect(defaultSelectDropdown).toHaveStyle('height: 36px;')
    })

    it('Size L >', () => {
      const { getByTestId } = renderSelect({ size: SelectSize.L })
      const defaultSelectDropdown = getByTestId(SELECT_TRIGGER_TEST_ID)

      expect(defaultSelectDropdown).toHaveStyle('height: 44px;')
    })

    it('Size XL >', () => {
      const { getByTestId } = renderSelect({ size: SelectSize.XL })
      const defaultSelectDropdown = getByTestId(SELECT_TRIGGER_TEST_ID)

      expect(defaultSelectDropdown).toHaveStyle('height: 54px;')
    })
  })
})
