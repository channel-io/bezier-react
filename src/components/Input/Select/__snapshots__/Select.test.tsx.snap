// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Test > Default Style > Snapshot > 1`] = `
.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin: 0px 0px 0px 6px;
  color: inherit;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
}

.c3 {
  font-size: 14px;
  line-height: 18px;
  margin: 0px 0px 0px 0px;
  font-style: normal;
  font-weight: normal;
  color: inherit;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
}

.c0 {
  position: relative;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 36px;
  padding: 8px 12px;
  color: #00000099;
  overflow: hidden;
  border-radius: 8px;
  box-shadow: 0 1px 2px #00000008, inset 0 0 0 1px #0000000D;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #FCFCFC;
  -webkit-transition-property: background-color,box-shadow;
  transition-property: background-color,box-shadow;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
}

.c1:hover {
  background-color: #F7F7F8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
  data-testid="bezier-react-select-container"
>
  <div
    class="c1"
    data-testid="bezier-react-select-trigger"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
        data-testid="bezier-react-text"
      />
    </div>
    <svg
      class="c4"
      data-testid="bezier-react-icon"
      fill="none"
      foundation="[object Object]"
      height="16"
      marginbottom="0"
      marginleft="6"
      marginright="0"
      margintop="0"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M19.707 8.793a1 1 0 010 1.414l-7 7a1 1 0 01-1.414 0l-7-7a1 1 0 011.414-1.414L12 15.086l6.293-6.293a1 1 0 011.414 0z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
